src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -903,25 +904,91 @@ private Point GetChildrenOffset()
 			return MeasureElement;
 		}
 
-		private (IDisposable Subscription, Memory<ViewPosition> Views) GetPositions()
+		private (IDisposable Subscription, Memory<ViewPosition> Views) GetPositions(int numberOfColumns, int numberOfRows)
 		{
+			Console.WriteLine($"GetPositions({numberOfColumns}, {numberOfRows}):");
+
 			var refs = Children.SelectToArray(c => (View: c, Handle: GCHandle.Alloc(c, GCHandleType.Normal)));
 
 			return (
 				Disposable.Create(() => refs.ForEach(c => c.Handle.Free())),
 				refs
 					.SelectToMemory(c =>
-						new ViewPosition(
-							c.Handle,
-							new GridPosition(
-								Grid.GetColumn(c.View),
-								Grid.GetRow(c.View),
-								Grid.GetColumnSpan(c.View),
-								Grid.GetRowSpan(c.View)
-							)
-						)
-					)
+					{
+						return MapViewToGridPosition(c);
+					})
 			);
+
+			ViewPosition MapViewToGridPosition((View View, GCHandle Handle) c)
+			{
+				var column = Grid.GetColumn(c.View);
+				var columnSpan = Grid.GetColumnSpan(c.View);
+
+				if (column < 0)
```

Comment:
MathEx.Clamp() is going to make this method a lot shorter ;) https://github.com/nventive/Uno/blob/f1ec504c0b5d0a681b234c5bbf8e5567cf61788a/src/Uno.UI/Extensions/MathEx.cs#L16

Reply:
I understand it would be better looking code.
But it's on a critical path and the most common scenario is "column == 1", so when it's happening I don't want to make any method call.

I'll move it to be the first one, btw.

Reply:
Is a non-virtual non-interroped method call really going to make any noticeable difference? Use `[MethodImpl(MethodImplOptions.AggressiveInlining)]` if you're really worried. The code is hideous as is.